### PR TITLE
Add ability to parse package metadata version 2.1

### PR DESCRIPTION
**Pull request by kihneman**
Fixes https://github.com/enthought/okonomiyaki/issues/310

### CODE REVIEW

**Comment by kihneman** [comment]
I still need to add tests. I tried generating actual 2.1 metdata for testing using `python setup.py bdist_egg`. However, the metadata always defaults to version 1.1, even if I add version 2.1 fields into the setup.py.

File: okonomiyaki/file_formats/_package_info.py [hunk 1]
```diff
@@ -220,7 +240,11 @@ def __init__(self, metadata_version, name, version, platforms=None,
...
         self.provides_dist = provides_dist or ()
         self.obsoletes_dist = obsoletes_dist or ()
-        self.project_urls = projects_urls or ()
+        self.project_urls = project_urls or ()
+
+        # version 2.1
+        self.description_content_type = description_content_type or ""
+        self.provides_extra = provides_extra or ()
 
     def to_string(self, metadata_version_info=MAX_SUPPORTED_VERSION):
```
Comment (itziakos) [inline okonomiyaki/file_formats/_package_info.py]: @kihneman We need to update the to_string method to be aware of PKG_INFO version 2.1

File: okonomiyaki/file_formats/_package_info.py [hunk 2]
```diff
@@ -231,10 +255,22 @@ def to_string(self, metadata_version_info=MAX_SUPPORTED_VERSION):
...
         self._write_field(s, 'Author', self.author)
         self._write_field(s, 'Author-email', self.author_email)
-        self._write_field(s, 'License', self.license)
+        if metadata_version_info >= (1, 2):
+            if self.maintainer:
+                self._write_field(s, 'Maintainer', self.maintainer)
+            if self.maintainer_email:
+                self._write_field(s, 'Maintainer-email', self.maintainer_email)
+
+        if self.license:
```
Comment (itziakos) [inline okonomiyaki/file_formats/_package_info.py]: Please always perform an explicit check for non boolean variables

File: okonomiyaki/file_formats/_package_info.py [hunk 3]
```diff
@@ -251,10 +287,27 @@ def to_string(self, metadata_version_info=MAX_SUPPORTED_VERSION):
         if metadata_version_info >= (1, 1):
             self._write_list(s, 'Classifier', self.classifiers)
 
+        if metadata_version_info >= (1, 2):
```
Comment (itziakos) [inline okonomiyaki/file_formats/_package_info.py]: relayout for clarity

```python
if metadata_version_info == (1, 1):
   ...
elif metadata_version_info >= (1, 2):
   ...
```

File: okonomiyaki/file_formats/_package_info.py [hunk 4]
```diff
@@ -176,6 +183,18 @@ def from_string(cls, s):
                     if value != 'UNKNOWN':
                         kw[attr_name] = value
 
+        if metadata_version_info >= (2, 1):
+            msg_body = msg.get_payload()
+            if msg_body is not None and msg_body != '':
+                if 'description' in kw:
+                    raise ValueError(
```
Comment (itziakos) [inline okonomiyaki/file_formats/_package_info.py]: I am not sure that we should raise an error here. Probably provide a warning. Let have an offline chat when time permits

File: okonomiyaki/file_formats/_package_info.py [hunk 5]
```diff
@@ -231,29 +255,69 @@ def to_string(self, metadata_version_info=MAX_SUPPORTED_VERSION):
...
-        self._write_field(s, 'License', self.license)
+
+        # Maintainer
+        if metadata_version_info >= (1, 2):
+            if self.maintainer:
+                self._write_field(s, 'Maintainer', self.maintainer)
+            if self.maintainer_email:
+                self._write_field(s, 'Maintainer-email', self.maintainer_email)
+
+        # License
```
Comment (itziakos) [inline okonomiyaki/file_formats/_package_info.py]: I would think that it is clear from the next few lines that this comment is stating the obvious. Please remove such comments from the `to_string` method

**Pull request merged**